packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeTokenLock.sol

```diff
@@ -0,0 +1,495 @@
+            hasBlockRequesterRoleOrIsManager(msg.sender),
+            ERROR_NOT_BLOCK_REQUESTER
+        );
+        require(_requesterAddress != address(0), ERROR_ZERO_ADDRESS);
+        airnodeToRequesterToBlockStatus[_airnode][_requesterAddress] = true;
+        emit BlockedRequester(_airnode, _requesterAddress, msg.sender);
+    }
+
+    /// @notice Function called to withdraw locked tokens of blocked requesters.
+    /// The tokens are transferred to the address specified in "blockWithdrawDestination"
+    /// @param _chainId The id of the chain
+    /// @param _airnode The airnode address
+    /// @param _endpointId The endpointId
+    /// @param _requesterAddress The address of the requester for which tokens are being withdrawn
+    /// @param _locker The address of the locker that locked tokens for the requester
+    function withdrawBlocked(
+        uint256 _chainId,
+        address _airnode,
+        bytes32 _endpointId,
+        address _requesterAddress,
+        address _locker
+    ) external override {
+        require(
+            airnodeToRequesterToBlockStatus[address(0)][_requesterAddress] ||
+                airnodeToRequesterToBlockStatus[_airnode][_requesterAddress],
+            ERROR_REQUESTER_NOT_BLOCKED
+        );
+        require(_chainId != 0, ERROR_ZERO_CHAINID);
+        require(_airnode != address(0), ERROR_ZERO_ADDRESS);
+        require(_requesterAddress != address(0), ERROR_ZERO_ADDRESS);
+        require(_locker != address(0), ERROR_ZERO_ADDRESS);
+        TokenLocks storage tokenLock = tokenLocks[_chainId][_airnode][
+            _endpointId
+        ][_requesterAddress];
+        require(tokenLock.lockerToLockAmount[_locker] != 0, ERROR_NOT_LOCKED);
+
+        uint256 amount = tokenLock.lockerToLockAmount[_locker];
+
+        tokenLock.lockerToLockAmount[_locker] = 0;
+        tokenLock.whitelistCount--;
+
+        if (tokenLock.whitelistCount == 0) {
+            IRequesterAuthorizerWithManager(
+                chainIdToRequesterAuthorizerWithManager[_chainId]
+            ).setIndefiniteWhitelistStatus(
+                    _airnode,
+                    _endpointId,
+                    _requesterAddress,
+                    false
+                );
+        }
+
+        assert(IERC20(api3Token).transfer(blockWithdrawDestination, amount));
+
+        emit WithdrewBlocked(
+            _chainId,
+            _airnode,
+            _endpointId,
+            _requesterAddress,
+            _locker,
+            blockWithdrawDestination,
+            amount
+        );
+    }
+
+    /// @notice Returns the current amount a locker has to
+    /// lock for a given chainId-airnode-endpointId
+    /// @param _chainId The id of the chain
+    /// @param _airnode The airnode address
+    /// @param _endpointId The endpointId
+    function getLockAmount(
+        uint256 _chainId,
+        address _airnode,
+        bytes32 _endpointId
+    ) public view override returns (uint256 _lockAmount) {
```

Comment:
This question I'm not sure about but do we need underscore for the return args as well?

Reply:
Hm probably not, Ill change it